pkg/codegen/python/python.go

```diff
@@ -119,8 +119,159 @@ func PyName(name string) string {
 				continue
 			}
 
-			// We want to fold digits immediately following an acronym into the same
-			// component as the acronym.
+			// We want to fold digits or the lowercase letter 's' immediately following
+			// an acronym into the same component as the acronym.
+			if unicode.IsDigit(char) || char == 's' {
+				// stateAcronym -> stateLowerOrNumber
+				state = stateLowerOrNumber
+				currentComponent.WriteRune(char)
+				continue
+			}
+
+			// stateAcronym -> stateLowerOrNumber
+			component := currentComponent.String()
+			last, size := utf8.DecodeLastRuneInString(component)
+			if result.Len() != 0 {
+				result.WriteRune('_')
+			}
+			result.WriteString(component[:len(component)-size])
+
+			currentComponent.Reset()
+			currentComponent.WriteRune(last)
+			currentComponent.WriteRune(char)
+			state = stateLowerOrNumber
+			continue
+
+		case stateLowerOrNumber:
+			if unicode.IsUpper(char) {
+				// stateLowerOrNumber -> stateUpper
+				if result.Len() != 0 {
+					result.WriteRune('_')
+				}
+				result.WriteString(currentComponent.String())
+
+				currentComponent.Reset()
+				currentComponent.WriteRune(unicode.ToLower(char))
+				state = stateUpper
```

Comment:
Should this be `unicode.ToUpper(char))`?

Reply:
This was from the original PyName algo and wasn't causing any issues. I cleaned it up and added some tests which all look good so I believe this is working as expected.